assets/src/components/v2/departures/section.ts

```diff
@@ -106,37 +123,88 @@ const trimOneBy = (
-    } = section;
-
-    for (let aboveIndex = aboveFold.length - 1; aboveIndex >= 0; aboveIndex--) {
-      const row = aboveFold[aboveIndex];
-
-      if (row.type == "departure_row") {
-        const trimmedTimeWithCrowding = _.last(row.times_with_crowding);
-
-        if (row.times_with_crowding.length > 1) {
-          // More than one departure time; trim the last one.
-          row.times_with_crowding.splice(row.times_with_crowding.length - 1);
-        } else {
-          // Only one departure time; trim this row entirely.
-          aboveFold.splice(aboveIndex, 1);
-        }
-
-        if (trimmedTimeWithCrowding != null) {
-          belowFold.unshift({
-            ...row,
-            id: _.uniqueId(`${row.id}_`),
-            times_with_crowding: [trimmedTimeWithCrowding],
-          });
-        }
-
-        return true;
+  const {
+    rows: { aboveFold, belowFold },
+  } = section;
+
+  for (let aboveIndex = aboveFold.length - 1; aboveIndex >= 0; aboveIndex--) {
+    const row = aboveFold[aboveIndex];
+
+    if (isDepartureRow(row)) {
+      const timesWithCrowding = row.times_with_crowding;
+      const trimmedTimeWithCrowding = _.last(timesWithCrowding);
+
+      if (timesWithCrowding.length > 1) {
+        // More than one departure time; trim the last one.
+        timesWithCrowding.splice(timesWithCrowding.length - 1);
+      } else {
+        // Only one departure time; trim this row entirely.
+        aboveFold.splice(aboveIndex, 1);
+      }
+
+      if (trimmedTimeWithCrowding != null) {
+        belowFold.unshift({
+          ...row,
+          id: _.uniqueId(`${row.id}_`),
+          times_with_crowding: [trimmedTimeWithCrowding],
+        });
       }
+
+      return true;
     }
   }
 
   return false;
 };
+
+const trimSectionForDestinationGrouping = (
+  section: FoldedDepartureGroupedSection,
+): boolean => {
+  const {
+    rows: { aboveFold },
+    direction_to_trim,
+  } = section;
+
+  // If there's only one direction, treat it as a timed list and trim the furthest time
+  const trimmedRow = trimRowForDestinationGrouping(
+    aboveFold,
+    direction_to_trim,
+  );
+
+  if (isDepartureRow(trimmedRow)) {
+    aboveFold.splice(aboveFold.indexOf(trimmedRow), 1);
+    return true;
+  }
+
+  return false;
+};
+
+const trimRowForDestinationGrouping = (
+  aboveFold: Row[],
+  directionToTrim: number,
+): Row | undefined => {
+  const directionOneRows = aboveFold.filter(
+    (row) => isDepartureRow(row) && row.direction_id == 1,
+  );
+  const directionZeroRows = aboveFold.filter(
+    (row) => isDepartureRow(row) && row.direction_id == 0,
+  );
+
+  if (directionZeroRows.length < directionOneRows.length) {
+    return _.last(directionOneRows);
+  }
+  if (directionZeroRows.length > directionOneRows.length) {
+    return _.last(directionZeroRows);
+  }
+
+  return directionToTrim == 1
+    ? _.last(directionOneRows)
+    : _.last(directionZeroRows);
+};
+
+const calculateDirectionToTrim = (rows: Row[]) => {
```

Comment:
nit: "calculate" is a word that doesn't add much meaning to a function name; it could be omitted here without losing anything, I think.